### PR TITLE
15769 - Affiliation Invitations - updates and fixes

### PR DESCRIPTION
**Pull request by ochiu**
https://github.com/bcgov/entity/issues/15769

*Description of changes:*

- update affiliation_invitations table: to_org_id and recipient_email is nullable (email is not required for passcode flow)
- update affiliation to be created for the from_org
- update to_org to be an optional parameter
- update auth check to not include business_identifier as there is no affiliation to the from_org when creating an invite
- update auth check to not include business_identifier when working with an affiliation invitation
- removed sent date parameter for creation (was already being set as part of create date, sent date updates when an email it resent)
- removed recipient email parameter for creation (this is fetched via the entity contacts), if there is no email a validation error is thrown
- recipient_email is masked when returned as part of an affiliation invitation
- moved email masking logic to utils
- introduced public schema for affiliation invitation (for use with masked email)
- update tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Review by seeker25 — APPROVED** [review]

File: auth-api/src/auth_api/utils/util.py
Comment (seeker25) [inline auth-api/src/auth_api/utils/util.py:79]: Minor `-> str`

**Comment by seeker25** [comment]
Waiting on approval from @hfekete

**Review by hfekete — APPROVED** [review]

**Pull request merged**
